### DISCUSSION TITLE
API: Cannot patch subscriber: "Subscriber type premium is not valid for this newsletter."

### DISCUSSION THREAD

**Discussion by christophberger** [buttondown/discussions#32] [Q&A] [answered]
Hi,

I am trying to set a test subscriber to premium, in order to test conditional email content. 

However, the API call returns an HTTP 400: "Subscriber type premium is not valid for this newsletter."

The newsletter is connected to Stripe, and a plan is set up (pay what you want). 

_Side note: Before trying the API, I tried to gift a subscriber a paid subscription, but that subscriber remains in status "gifted". (No emails to that subscriber, no change of status.)_

What am I doing wrong? Can I make a subscriber a premium subscriber through the API? Or is "gifted" the best I can get without actually having the test subscriber enter payment details with Stripe?

```
PATCH /v1/subscribers/8136be14-c95d-423c-ae83-e45da5ca54bf HTTP/1.1
Content-Type: application/json
Accept: application/json
Authorization: Token <REDACTED>
User-Agent: PostmanRuntime/7.37.0
Cache-Control: no-cache
Host: api.buttondown.email
Accept-Encoding: gzip, deflate, br
Connection: keep-alive
Content-Length: 36
Cookie: initial_path="/v1/ping"
 
{
"subscriber_type": "premium"
}
 
HTTP/1.1 400 Bad Request

(...)
 
{"code":"subscriber_type_invalid","detail":"Subscriber type premium is not valid for this newsletter.","metadata":{}}
```

**Comment by jmduke** [comment, marked as answer]
Yup, your understanding is exactly correct — you can't mark a subscriber as premium 'manually', that has to be governed by Stripe. You'll want to mark them as gifted instead.

(I'll tweak the API response to make that more explicit!)

**Comment by christophberger** [comment]
Thank you, @jmduke. 
I just found out that `subscriber.can_view_premium_content` evaluates to `true` for status `gifted`, too, which is what I need. So I would not need a `premium` subscriber for testing.
